src/context/Authentication/AuthProvider.js

```diff
@@ -23,35 +24,37 @@ const AuthState = ({ children }) => {
       password,
     }).then((res) => {
       if (res.status === 200) {
-        AsyncStorage.setItem('token', res.data.access_token);
+        storage.set("token", res.data.access_token);
         dispatch({
-          type: 'GET_TOKEN',
+          type: "GET_TOKEN",
           token: res.data.access_token,
```

Comment:
You are singing in the user for the first time, not only getting the token. I think you should set the user directly after getting the token successfully.